### PR TITLE
Fixed Factory construct just for calling .Name()

### PR DESCRIPTION
**Pull request by rockeet**

### CODE REVIEW

File: options/options_helper.cc [hunk 1]
```diff
@@ -939,7 +939,7 @@ Status GetTableFactoryFromMap(
     }
     table_factory->reset(new BlockBasedTableFactory(bbt_opt));
     return s;
-  } else if (factory_name == PlainTableFactory().Name()) {
+  } else if (factory_name == "PlainTable") {
```
Comment (ajkr) [inline options/options_helper.cc:942]: Can we have a named constant here too?

Comment (rockeet) [reply options/options_helper.cc:942]: > Can we have a named constant here too?

No, there is no such a constant defined for PlainTable.

Comment (ajkr) [reply options/options_helper.cc:942]: How about define such a constant for "PlainTable" first, then?

Comment (rockeet) [reply options/options_helper.cc]: > How about define such a constant for "PlainTable" first, then?

OK, done!

**Pull request closed without merging**